semver.go

```diff
@@ -11,32 +12,78 @@ const (
 	numbers  string = "0123456789"
 	alphas          = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ-"
 	alphanum        = alphas + numbers
+
+	// threeFourDigitVersionRegexFormat represents the strict rules for parsing 3 and 4 digit
+	// version numbers. There are six capture groups which define major, minor, patch, revision,
+	// pre-release, and build metadata. When using 3 digit versioning strict semver adherence
+	// will be utilized/
+	threeFourDigitVersionRegexFormat = `^(?P<major>0|[1-9]\d*)\.` +
+		`(?P<minor>0|[1-9]\d*)\.` +
+		`(?P<patch>0|[1-9]\d*)` +
+		`(?:\.(?P<revision>0|[1-9]\d*))?` +
+		`(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?` +
+		`(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$`
+
+	// threeFourDigitTolerantVersionRegexFormat represents loose rules for parsing 3 and 4 digit
+	// version numbers. There are six capture groups which define major, minor, patch, revision,
+	// pre-release, and build metadata. The loose format allows for leading and trailing spaces,
+	// prefixing versions with a 'v' and empty capture groups in order to dynamically fill in the
+	// fields with defaults. When using 3 digit versioning loose semver adherence will be utilized.
+	threeFourDigitTolerantVersionRegexFormat = `^(?:\s*)?(?:v)?` +
```

Comment:
Maybe using `Loose` instead of `Tolerant`? The latter doesn't sound right, but maybe it's because of my not-native-english ears :P

Reply:
Indeed. I chose tolerant because of the `ParseTolerant()` method call, but I think that makes more sense as the function name and action that it performs; loose or some form of that does sound more appealing.

Reply:
Updating the name to permissive and also updating the documentation to give more context.